src/Microsoft.Framework.Configuration/Internal/ConfigurationFocus.cs

```diff
@@ -6,77 +6,95 @@
 
 namespace Microsoft.Framework.Configuration.Internal
 {
-    public class ConfigurationFocus : IConfiguration
+    public class ConfigurationFocus : IConfigurationSection
     {
+        private string _key { get; set; }
+        private string _value { get; set; }
+
         private readonly string _prefix;
-        private readonly IConfiguration _root;
+        private readonly IConfigurationSection _root;
```

Comment:
If this is the immediate parent, it should be renamed and be of type `IConfiguration`. If it is actuallythe root then it should probably be `IConfigurationRoot`